moabb/evaluations/metasplitters.py

```diff
@@ -0,0 +1,179 @@
+import numpy as np
+from sklearn.model_selection import (
+    BaseCrossValidator,
+    StratifiedShuffleSplit,
+)
+
+
+class OfflineSplit(BaseCrossValidator):
+
+    def __init__(self, n_folds):
+        self.n_folds = n_folds
+
+    def get_n_splits(self, metadata):
+        subjects = len(metadata.subject.unique())
+        sessions = len(metadata.session.unique())
+        return subjects * sessions
+
+    def split(self, X, y, metadata, **kwargs):
+
+        subjects = metadata.subject
+
+        for subject in subjects.unique():
+            X_, y_, meta_ = (
+                X[subjects == subject],
+                y[subjects == subject],
+                metadata[subjects == subject],
+            )
+            sessions = meta_.session.values
+
+            for session in sessions:
+                ix_test = np.nonzero(sessions == session)[0]
+
+                yield ix_test
+
+
+class TimeSeriesSplit(BaseCrossValidator):
+
+    def __init__(self, calib_size):
+        self.calib_size = calib_size
+
+    def get_n_splits(self, metadata):
+        sessions = metadata.session.unique()
+        subjects = metadata.subject.unique()
+
+        splits = len(sessions) * len(subjects)
+        return splits
+
+    def split(self, X, y, metadata, **kwargs):
+
+        runs = metadata.run.unique()
+        sessions = metadata.session.unique()
+        subjects = metadata.subject.unique()
+
+        if len(runs) > 1:
+            for subject in subjects:
+                for session in sessions:
+                    # Index of specific session of this subejct
+                    session_indices = metadata[
+                        (metadata.subject == subject) & (metadata.session == session)
+                    ].index
+
+                    for run in runs:
+                        test_ix = session_indices[
+                            metadata.loc[session_indices].run != run
+                        ]
+                        calib_ix = session_indices[
+                            metadata.loc[session_indices].run == run
+                        ]
+                        yield test_ix, calib_ix
+                        break  # Take the fist run as calibration
```

Comment:
The list of runs will not necessarily be in the correct order. All run and session names start with an index (supposed to) indicate their order and eventually followed by an additional description. You can get these using the following:
```python
import re
pattern = r"([0-9]+)(|[a-zA-Z]+[a-zA-Z0-9]*)”
session_name = “0train”
index, description = re.fullmatch(pattern, session_name).groups()
index = int(index)
print(index, description) # 0, train
```
Then, you will be able to sort the runs according to their index :)